plugins/modules/ndo_tenant_span_session.py

```diff
@@ -0,0 +1,624 @@
+                enforce_span_version=dict(type="bool"),
+                flow_id=dict(type="int"),
+                ttl=dict(type="int"),
+                dscp=dict(type="str", choices=list(TARGET_DSCP_MAP)),
+            ),
+        ),
+        state=dict(type="str", default="query", choices=["absent", "query", "present"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        mutually_exclusive=[
+            ("template", "template_id"),
+        ],
+        required_if=[
+            ["state", "absent", ["name", "uuid"], True],
+            ["state", "present", ["name", "uuid"], True],
+        ],
+        required_one_of=[["template", "template_id"]],
+    )
+
+    mso = MSOModule(module)
+    mso_schema = None
+
+    template_name = module.params.get("template")
+    template_id = module.params.get("template_id")
+    name = module.params.get("name")
+    uuid = module.params.get("uuid")
+    description = module.params.get("description")
+    sources = module.params.get("sources")
+    admin_state = module.params.get("admin_state")
+    mtu = module.params.get("mtu")
+    destination_epg = module.params.get("destination_epg")
+    state = module.params.get("state")
+
+    ops = []
+    match = None
+
+    mso_template = MSOTemplate(mso, "monitoring_tenant", template_name, template_id)
+    mso_template.validate_template("monitoring")
+    object_description = "SPAN Session"
+    path = "/monitoringTemplate/template/spanSessions"
+    span_session_path = None
+    reference_collection = {
+        "epg": {
+            "name": "epgName",
+            "reference": "epgRef",
+            "type": "epg",
+            "template": "epgTemplateName",
+            "templateId": "epgTemplateId",
+            "schema": "epgSchemaName",
+            "schemaId": "epgSchemaId",
+        }
+    }
+
+    existing_span_sessions = mso_template.template.get("monitoringTemplate", {}).get("template", {}).get("spanSessions") or []
+
+    if name or uuid:
+        match = mso_template.get_object_by_key_value_pairs(
+            object_description,
+            existing_span_sessions,
+            [KVPair("uuid", uuid) if uuid else KVPair("name", name)],
+        )
+        if match:
+            span_session_path = "{0}/{1}".format(path, match.index)
+            set_template_and_references(mso_template, match.details, reference_collection)
+            mso.existing = mso.previous = copy.deepcopy(match.details)
+    else:
+        mso.existing = mso.previous = [set_template_and_references(mso_template, obj, reference_collection) for obj in existing_span_sessions]
+
+    if state == "present":
+        if uuid and not mso.existing:
+            mso.fail_json(msg="{0} with the UUID: '{1}' not found".format(object_description, uuid))
+
+        destination_epg_uuid = get_epg_uuid(mso, mso_schema, destination_epg.get("epg"), destination_epg.get("epg_uuid"))
```

Comment:
You need to add `destination_epg is not None` before using `destination_epg.get`.  Because the `destination_epg` is optional when we do the update.

Reply:
Fixed